### DISCUSSION TITLE
External Plugins in Kustomize Build

### DISCUSSION THREAD

**Discussion by SachithKasthuriarachchi** [fluxcd/flux2#3819] [Q&A] [answered]
Does flux support using external Kustomize plugins within the `kustomize build`? I have developed a hasher plugin which appends a hash to names of the secretproviderclasses and updates the references in Deployments, StatefulSets etc. Is there a way for me to incorporate this plugin with flux?

Basically, I want to incorporate `--enable-alpha-plugins` to `kustomize build` in flux

**Comment by SachithKasthuriarachchi** [comment]
The [docs](https://fluxcd.io/flux/faq/#what-is-the-behavior-of-kustomize-used-by-flux) says this is disabled by default. Is there a way to enable it?

**Comment by SachithKasthuriarachchi** [comment]
@stefanprodan is there a way to do this?

**Reply by stefanprodan** [reply, marked as answer]
No and probably never will. Flux v2 was designed with a security first approach where we guarantee that Flux doesn’t shell exec nor runs any untrusted code.

**Reply by SachithKasthuriarachchi** [reply]
ack, thanks for the clarification.

**Reply by stefanprodan** [reply]
For custom YAML generators I suggest looking into https://fluxcd.io/flux/cheatsheets/oci-artifacts/

**Reply by SachithKasthuriarachchi** [reply]
hi @stefanprodan I believe the idea is to feed flux with a generated output yaml. Please correct me if I'm wrong

**Reply by stefanprodan** [reply]
Yes, you would run in CI something like `kustomize build --enable-alpha-plugins . | flux push artifact -f-`, then use an OCIRepository instead of Git in your Flux Kustomization.

**Comment by towens** [comment]
Too bad. ksops is wonderful.

**Reply by stefanprodan** [reply]
Flux has native SOPS support: https://fluxcd.io/flux/components/kustomize/kustomizations/#decryption

**Reply by towens** [reply]
Yes, thanks for the native SOPS support. I switched our secrets to it yesterday.
